Classes/Command/BackupCommandController.php

```diff
@@ -302,28 +308,48 @@ protected function legacyRestore($tmpFolder, $fileBackups) {
      */
     protected function createCopyOfTablesToMerge()
     {
-        $db = $this->getDatabaseConnection();
+        /** @var Connection $dbConnection */
+        $dbConnection = GeneralUtility::makeInstance(ConnectionPool::class)->getConnectionByName(ConnectionPool::DEFAULT_CONNECTION_NAME);
         foreach (['sys_domain', 'be_users', 'tx_scheduler_task'] as $table) {
-            $db->sql_query('CREATE TABLE ' . $table . '_local LIKE ' . $table);
-            $db->sql_query('INSERT INTO ' . $table . '_local SELECT * FROM ' . $table);
+            try {
+                $dbConnection->exec('CREATE TABLE ' . $table . '_local LIKE ' . $table);
+                $dbConnection->exec('INSERT INTO ' . $table . '_local SELECT * FROM ' . $table);
+            } catch (DBALException $ex) {
+                $this->output->outputLine('<error>Failed creating</error> <i>%s</i>', [$table . '_local']);
+                return;
+            }
             $this->output->outputLine('Created <i>%s</i>', [$table . '_local']);
         }
     }
 
+    protected function dropCopiedTablesUsedForMerge()
+    {
+        /** @var Connection $dbConnection */
+        $dbConnection = GeneralUtility::makeInstance(ConnectionPool::class)->getConnectionByName(ConnectionPool::DEFAULT_CONNECTION_NAME);
+        foreach (['sys_domain', 'be_users', 'tx_scheduler_task'] as $table) {
+            try {
+                $dbConnection->exec('DROP TABLE  ' . $table . '_local');
+                $this->output->outputLine('<info>Deleted</info> <i>%s</i>', [$table . '_local']);
+            } catch (DBALException $ex) {
+                $this->output->outputLine('<error>Failed dropping table </error> <i>%s</i>', [$table . '_local']);
+            }
+        }
+    }
+
     /**
      * Sanitize restored tables
      *
      * @todo: make it possible to extend this through configuration
      */
     protected function sanitizeRestoredTables()
     {
-        $db = $this->getDatabaseConnection();
-
-        $availableTables = $db->admin_get_tables();
-        if (isset($availableTables['fe_users'])) {
-            $db->sql_query('UPDATE fe_users SET username = MD5(username), password = MD5(password), email = CONCAT(LEFT(UUID(), 8), "@beech.it") WHERE email NOT LIKE "%@beech.it"');
-
+        /** @var Connection $connection */
+        $connection = GeneralUtility::makeInstance(ConnectionPool::class)->getConnectionForTable('fe_users');
+        try {
+            $connection->exec('UPDATE fe_users SET username = MD5(username), password = MD5(password), email = CONCAT(LEFT(UUID(), 8), "@beech.it") WHERE email NOT LIKE "%@beech.it"');
```

Comment:
Why not use the query builder here to?

Reply:
The MD5 could not be used. therefore this can be ignored